frontends/web/src/routes/buy/utils.tsx

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright 2022 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ExchangeDealsList } from '../../api/exchanges';
+
+/**
+ * Finds the lowest fee among all `supported`
+ * exchange providers for a given region.
+ */
+export function findLowestFee(providers: ExchangeDealsList) {
+  // all payment methods' fees of all
+  // supported providers (for a selected region)
+  let allFees: number[] = [];
+
+  providers.exchanges.forEach(provider => {
+    provider.deals.forEach(deal => {
+      if (provider.supported) {
+        allFees = [...allFees, deal.fee];
+      }
+    });
+  });
+
+  return Math.min(...allFees);
+}
+
+
+export function findBestDeal(providers: ExchangeDealsList, lowestFee: number): ExchangeDealsList {
+  // for each provider's payment method
+  // mark it as "bestDeal" if the
+  // fee is equal to the lowest fee
+  // accross ALL supported providers
+  const hasMultipleSupportedExchanges = providers.exchanges.filter(p => p.supported).length > 1;
+  const exchanges = providers.exchanges.map(exchange => ({
```

Comment:
Did you mean if it's `false`? Anyhow, would you be able to suggest the code change? I don't think I understood you 😅 

What this map is doing is to mark `isBestDeal` is `true` if it's the lowest fee across all providers AND only when there's more than one exchanges. Otherwise it's `false`. Regardless, we want `isBestDeal` to always be there. Because before this .map(), isBestDeal is still not there yet.

Reply:
> Did you mean if it's `false`? Anyhow, would you be able to suggest the code change? I don't think I understood you sweat_smile
> 
> What this map is doing is to mark `isBestDeal` is `true` if it's the lowest fee across all providers AND only when there's more than one exchanges. Otherwise it's `false`. Regardless, we want `isBestDeal` to always be there. Because before this .map(), isBestDeal is still not there yet.

Ahah right :D I wrote that before suggesting the type changes! Not valid anymore I guess :sweat_smile: